src/ResourceManager/Resources/Commands.Resources/RoleAssignments/ValidateScope.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.Common
+{
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+    public class ValidateScope : ValidateArgumentsAttribute
+    {
+        protected override void Validate(object arguments, EngineIntrinsics engineIntrinsics)
+        {
+            if (arguments != null)
+            {
+                string scope = (string)arguments;
+
+                if (scope.Length == 0 || !scope.StartsWith("/"))
+                {
+                    throw new ValidationMetadataException("Scope must begin with '/'.");
+                }
+
+                // "/" is a valid scope.
+                if (string.Compare(scope, "/") == 0)
+                {
+                    return;
+                }
+
+                // For scope that on tenant level, we don't perform further check here.
+                if (!scope.StartsWith("/subscriptions", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return;
+                }
```

Comment:
we should validate that scope either starts with subscriptions or providers. and for the providers scenario, as we discussed we should check that the no. of parts is even.